src/app/page.tsx

```diff
@@ -82,6 +90,10 @@ export default async function Main({ searchParams }: { searchParams: { token: st
               'use server'
               await updateViewModeSettings(token, mode)
             }}
+            getTokenPayload={async (): Promise<Token> => {
+              'use server'
+              return getTokenPayload(token)
+            }}
```

Comment:
We don't need to use a server action like this here. Page should already be a server component. Why don't we use `getTokenPayload` directly in the server component (outside the return) and use its values as props instead?